projects/plugins/jetpack/extensions/blocks/premium-content/premium-content.php

```diff
@@ -152,9 +152,17 @@ function stripe_nudge( $checkout_url, $description, $button_text ) {
  */
 function add_paid_content_post_meta( int $post_id, WP_Post $post ) {
 	$contains_paid_content = has_block( 'premium-content/container', $post );
-	update_post_meta(
-		$post_id,
-		META_NAME_CONTAINS_PAID_CONTENT,
-		$contains_paid_content
-	);
+	if ( $contains_paid_content ) {
+		update_post_meta(
+			$post_id,
+			META_NAME_CONTAINS_PAID_CONTENT,
+			$contains_paid_content
+		);
+	}
+	if ( ! $contains_paid_content ) {
```

Comment:
I'm intrigued. Any particular reason why you did not user `else`? It's perfectly fine, but I wanted to know what drove this decision...

Reply:
This was something @sirbrillig taught me on Shilling. By making every logic blog independent, you force people to think through the actual logic being checked. Bugs often get introduced by people thinking through what's in the if statement but not thinking through what's in the else statement. Payton can probably elaborate much better.

Reply:
I knew @sirbrillig was involved :)
That's a good answer, no need to elaborate. I was just being curious, and know I will probably start to use that in my code! Thanks!

Reply:
😁 For reference: https://payton.codes/2019/01/22/alternatives-to-else/